internal/pkg/runtime/launcher/oci/launcher_linux.go

```diff
@@ -397,9 +403,26 @@ func (l *Launcher) finalizeSpec(ctx context.Context, b ocibundle.Bundle, spec *s
 		return err
 	}
 
+	l.handleTmpVarTmpSymlink(spec)
+
 	return b.Update(ctx, spec)
 }
 
+func (l *Launcher) handleTmpVarTmpSymlink(spec *specs.Spec) {
+	tmpResolved := fs.EvalRelative(tmpPath, spec.Root.Path)
+	varTmpResolved := fs.EvalRelative(vartmpPath, spec.Root.Path)
+	sylog.Debugf("Container /tmp resolves to %q", tmpResolved)
+	sylog.Debugf("Container /var/tmp resolves to %q", varTmpResolved)
+
+	if varTmpResolved != tmpResolved {
+		return
+	}
+
+	spec.Mounts = lo.Filter(spec.Mounts, func(m specs.Mount, _ int) bool {
+		return (m.Destination != vartmpPath)
```

Comment:
Sorry - thinking about this implementation is giving me pause for thought again. r.e. explicit binds of both `-B /tmp` and `-B /var/tmp` where there is a symlink.

This filtering here is different than the native mode, because the native mode exclusion of `/var/tmp` is only for the automatic mount, not a user requested bind.

I need to go back and look at Docker / Podman behaviours etc. and get it straight in my head. It might be correct as-is.... apologies for delaying this. My fault!

Reply:
I've thought through this now, and looked at Docker in the gist here: https://gist.github.com/dtrudg/4dda5e5e2d3c01fa00ae1f040a5ebeb6

We can't do a simple filter by destination for anything mounted onto `/var/tmp`, as this removes both the default and any explicit user-requested `/var/tmp` binds... which must be handled differently:

The default mount onto `/var/tmp` must be skipped if `/var/tmp` is a symlink to `/tmp`. This is mostly important in OCI-mode in conjunction with `--workdir`... to ensure `workdir/tmp` and not `workdir/var_tmp` is present at `/tmp` in the container.

If a user requests `-B /var/tmp` then it must be honoured, and never filtered out, even if it is going to mask something else on `/tmp` via the symlink.